File: MechanicalSheets.Api/Models/Sheet.cs
using MechanicalSheets.Api.Enums;

namespace MechanicalSheets.Api.Models;

public class Sheet
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string? Brand { get; set; }
    public string? Vehicle { get; set; }
    public DateOnly InspectionDate { get; set; }
    public SheetStatusEnum SheetStatus { get; set; } = SheetStatusEnum.Draft;
    public string? RejectionNote { get; set; }
    public DateTime? SubmittedAt { get; set; }
    public DateTime? ReviewedAt { get; set; }
    public int CreatedById { get; set; }
    public int? ReviewedById { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    
    public User CreatedBy { get; set; } = null!;
    public User? ReviewedBy { get; set; }
    public ICollection<SheetTechnician> Technicians { get; set; } = new List<SheetTechnician>();
    public ICollection<SheetDefectItem> DefectItems { get; set; } = new List<SheetDefectItem>();
}

File: frontend-angular/src/app/models/sheet.ts
import { SheetStatus } from './sheet-status';
import { DefectItemResponse } from './defect-item';
import { UserSummary } from './user';

// Specchio di SheetResponseDto.cs. InspectionDate = DateOnly → string 'YYYY-MM-DD'.
export interface SheetResponse {
  id: number;
  code: string;
  brand?: string | null;
  vehicle?: string | null;
  inspectionDate: string;
  status: SheetStatus;
  rejectionNote?: string | null;
  submittedAt?: string | null;
  reviewedAt?: string | null;
  createdAt: string;
  createdBy: UserSummary;
  reviewedBy?: UserSummary | null;
  technicians: UserSummary[];
  defectItems: DefectItemResponse[];
}

// Specchio di CreateSheetDto.cs — TechnicianIds obbligatorio (anche vuoto).
export interface CreateSheetDto {
  code: string;
  brand?: string | null;
  vehicle?: string | null;
  inspectionDate: string;
  technicianIds: number[];
}

// Specchio di UpdateSheetDto.cs — tutto opzionale, solo Draft/Rejected.
export interface UpdateSheetDto {
  code?: string;
  brand?: string | null;
  vehicle?: string | null;
  inspectionDate?: string;
  technicianIds?: number[];
}

// Specchio di RejectSheetDto.cs — nota obbligatoria lato UI.
export interface RejectSheetDto {
  rejectionNote: string;
}
